Sources/Contour/Transport/RabbitMQ/Internal/ProducerRegistry.cs

```diff
@@ -77,46 +72,57 @@ public void Reset()
+
+            var producer = this.TryResolverFor(configuration.Label);
             if (producer == null)
             {
-                using (RabbitChannel channel = this._bus.OpenChannel())
-                {
-                    var topologyBuilder = new TopologyBuilder(channel);
-                    var builder = new RouteResolverBuilder(this._bus.Endpoint, topologyBuilder, configuration);
-                    Maybe<Func<IRouteResolverBuilder, IRouteResolver>> routeResolverBuilder = configuration.Options.GetRouteResolverBuilder();
+                var topologyBuilder = new TopologyBuilder(connection.OpenChannel());
+                var builder = new RouteResolverBuilder(this.bus.Endpoint, topologyBuilder, configuration);
+                var routeResolverBuilder = configuration.Options.GetRouteResolverBuilder();
 
-                    Assumes.True(routeResolverBuilder.HasValue, "RouteResolverBuilder must be set for [{0}]", configuration.Label);
+                Assumes.True(
+                    routeResolverBuilder.HasValue,
+                    "RouteResolverBuilder must be set for [{0}]",
+                    configuration.Label);
 
-                    IRouteResolver routeResolver = routeResolverBuilder.Value(builder);
+                var routeResolver = routeResolverBuilder.Value(builder);
 
-                    producer = new Producer(this._bus, configuration.Label, routeResolver, configuration.Options.IsConfirmationRequired());
-                    producer.Failed += p =>
-                        {
-                            {
-                                // lock (_producers)
-                                p.Dispose();
-                                this._producers.Remove(p.Label);
-                            }
-                        };
+                producer = new Producer(
+                    this.bus.Endpoint,
+                    connection,
+                    configuration.Label,
+                    routeResolver,
+                    configuration.Options.IsConfirmationRequired());
 
-                    // lock (_producers)
-                    this._producers.Add(configuration.Label, producer);
-                }
-
-                if (configuration.RequiresCallback)
+                producer.Failed += p =>
                 {
-                    producer.UseCallbackListener(this._bus.ListenerRegistry.ResolveFor(configuration.CallbackConfiguration));
-                }
+                    {
+                        p.Dispose();
+                        this.producers.Remove(p.Label);
+                    }
+                };
+
+                this.producers.Add(configuration.Label, producer);
+            }
+
+            if (configuration.RequiresCallback)
+            {
+                producer.UseCallbackListener(
+                    this.bus.ListenerRegistry.ResolveFor(configuration.CallbackConfiguration));
```

Comment:
Need a test for this case to check: receiving using con string 1 and producing requests using con string 2. See https://github.com/SDVentures/Contour/issues/17